test/specs/dom/index.js

```diff
@@ -0,0 +1,545 @@
+/** only run DOM tests in browsers */
+if (typeof window !== "undefined") {
+  const PalindromDOM = require("../../../src/palindrom-dom");
+  const assert = require("assert");
+  const moxios = require("moxios");
+  const sinon = require("sinon");
+  const expect = require("chai").expect;
+
+  function createLinkTest(href, parent) {
+    parent = parent || document.body;
+    const a = document.createElement("A");
+    a.innerHTML = "Link";
+    a.href = href;
+    parent.appendChild(a);
+    parent.addEventListener("click", clickHandler);
+    fireEvent(a, "click");
+    parent.removeEventListener("click", clickHandler);
+    parent.removeChild(a);
+  }
+  function createLinkTestNested(href, parent) {
+    parent = parent || document.body;
+    const a = document.createElement("A");
+    a.innerHTML = "<strong>Link</strong>";
+    a.href = href;
+    parent.appendChild(a);
+    parent.addEventListener("click", clickHandler);
+    fireEvent(a.firstChild, "click");
+    parent.removeEventListener("click", clickHandler);
+    parent.removeChild(a);
+  }
+
+  function fireEvent(fireOnThis, evt) {
```

Comment:
We no longer need to trigget any event for dirty checking, right? So is this just to test the PalindromDOM behavior for morphing URL? I'd rename to to the "clickElement" as this is the only use-case